packages/editor/example/index.ts

```diff
@@ -28,12 +29,52 @@ window.onload = () => {
     };
   };
 
+  const getNamingFunction =
+    (schema: JsonSchema) => (element: Object): string => {
+
+      if (!_.isEmpty(labelProvider) && labelProvider[schema.id] !== undefined) {
+
+        if (typeof labelProvider[schema.id] === 'string') {
+          // To be backwards compatible: a simple string is assumed to be a property name
+          return element[labelProvider[schema.id]];
+        }
+        if (typeof labelProvider[schema.id] === 'object') {
+          const info =  labelProvider[schema.id] as LabelDefinition;
+          let label;
+          if (info.constant !== undefined) {
+            label = info.constant;
+          }
+          if (!_.isEmpty(info.property) && !_.isEmpty(element[info.property])) {
+            label = _.isEmpty(label) ?
+              element[info.property] : `${label} ${element[info.property]}`;
+          }
+          if (label !== undefined) {
+            return label;
+          }
+        }
+      }
+      const namingKeys =
+        Object.keys(schema.properties).filter(key => key === 'id' || key === 'name');
+      if (namingKeys.length !== 0) {
+        return element[namingKeys[0]];
+      }
+
+      return JSON.stringify(element);
+    };
+
+  const imageGetterPredicate = (schemaId: string) =>
+    !_.isEmpty(imageProvider) ? `icon ${imageProvider[schemaId]}` : '';
+
   const store = createEditorStore({}, taskSchema, uischema, materialFields,
                                   materialRenderers, imageProvider, labelProvider, modelMapping,
                                   detailSchemata, {});
 
   ide.filterPredicate = filterPredicate;
 
+  ide.namingPredicate = getNamingFunction;
```

Comment:
These aren't predicates since they don't return a boolean, I'd rather stick with naming schema of `labelProvider` and `imageProvider` throughout the code.